### PR TITLE
[YUNIKORN-1888] Allow allocations to be added to existing nodes

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Adds the ability to assign existing allocations dynamically to an already created node. This allows "late" events to be processed properly.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [x] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1888

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/context.go [hunk 1]
```diff
@@ -765,6 +769,71 @@ func (cc *ClusterContext) handleRMUpdateAllocationEvent(event *rmevent.RMUpdateA
...
 }
 
+func (cc *ClusterContext) processAllocations(request *si.AllocationRequest) {
+	// Send rejected allocations back to RM
+	rejectedAllocs := make([]*si.RejectedAllocation, 0)
+
+	// Send to scheduler
+	for _, siAlloc := range request.Allocations {
+		// try to get ApplicationInfo
+		partition := cc.GetPartition(siAlloc.PartitionName)
```
Comment (wilfred-s) [inline pkg/scheduler/context.go:779]: We're getting the partition here

File: pkg/scheduler/context.go [hunk 2]
```diff
@@ -765,6 +769,71 @@ func (cc *ClusterContext) handleRMUpdateAllocationEvent(event *rmevent.RMUpdateA
...
+			rejectedAllocs = append(rejectedAllocs, &si.RejectedAllocation{
+				AllocationKey: siAlloc.AllocationKey,
+				ApplicationID: siAlloc.ApplicationID,
+				Reason:        msg,
+			})
+			continue
+		}
+
+		// try adding to app
+		node := cc.GetNode(siAlloc.GetNodeID(), siAlloc.GetPartitionName())
```
Comment (wilfred-s) [inline pkg/scheduler/context.go:796]: This already happens when we call `partition.addAllocation()`

**Review by wilfred-s — CHANGES_REQUESTED** [review]

**Comment by craigcondit** [comment]
Rebased PR on master after merging #615 and addressed review comments.

File: pkg/scheduler/context.go (hunk 2 above)
Comment (craigcondit) [reply pkg/scheduler/context.go]: Correct, this was here as we needed the node instance type to construct the real alloc from the SI alloc. Now that #615 is merged this has been removed.

**Pull request closed without merging**